Runtime/ApiRequests.cs

```diff
@@ -109,16 +125,129 @@ public static async Task DownloadFileAsync(string url, string path)
             }
             else
             {
-                return await GetAsync<SkyboxTip>("skybox/get-one-tip", apiKey);
+                return await GetAsync<SkyboxTip>("skybox/get-one-tip");
+            }
+        }
+
+        public static async Task<GetHistoryResult> GetSkyboxHistoryAsync(HistorySearchQueryParameters searchQueryParams = null)
+        {
+            Dictionary<string, string> searchQuery = null;
+
+            if (searchQueryParams != null)
+            {
+                searchQuery = new Dictionary<string, string>();
+
+                if (!string.IsNullOrWhiteSpace(searchQueryParams.StatusFilter))
+                {
+                    searchQuery.Add("status", searchQueryParams.StatusFilter.ToLower());
+                }
+
+                if (searchQueryParams.Limit.HasValue)
+                {
+                    searchQuery.Add("limit", searchQueryParams.Limit.ToString());
+                }
+
+                if (searchQueryParams.Offset.HasValue)
+                {
+                    searchQuery.Add("offset", searchQueryParams.Offset.ToString());
+                }
+
+                if (!string.IsNullOrWhiteSpace(searchQueryParams.Order))
+                {
+                    searchQuery.Add("order", searchQueryParams.Order.ToUpper());
+                }
+
+                if (searchQueryParams.ImagineId.HasValue)
+                {
+                    searchQuery.Add("imagine_id", searchQueryParams.ImagineId.ToString());
+                }
+
+                if (!string.IsNullOrWhiteSpace(searchQueryParams.QueryFilter))
+                {
+                    searchQuery.Add("query", UnityWebRequest.EscapeURL(searchQueryParams.QueryFilter));
+                }
+
+                if (!string.IsNullOrWhiteSpace(searchQueryParams.GeneratorFilter))
+                {
+                    searchQuery.Add("generator", UnityWebRequest.EscapeURL(searchQueryParams.GeneratorFilter));
+                }
+
+                if (searchQueryParams.FavoritesOnly.HasValue &&
+                    searchQueryParams.FavoritesOnly.Value)
+                {
+                    searchQuery.Add("my_likes", searchQueryParams.FavoritesOnly.Value.ToString().ToLower());
+                }
+
+                if (searchQueryParams.GeneratedBy.HasValue)
+                {
+                    searchQuery.Add("api_key_id", searchQueryParams.GeneratedBy.Value.ToString());
+                }
+
+                if (searchQueryParams.SkyboxStyleId != 0)
+                {
+                   searchQuery.Add("skybox_style_id", searchQueryParams.SkyboxStyleId.ToString());
+                }
             }
+
+            return await GetAsync<GetHistoryResult>("imagine/myRequests", searchQuery);
+        }
+
+        public static async Task<ImagineResult> ToggleFavorite(int imagineId)
+        {
+#if UNITY_2022_1_OR_NEWER
+            using var request = UnityWebRequest.PostWwwForm($"{ApiEndpoint}imagine/favorite/{imagineId}", null);
```

Comment:
I'm curious why we don't just always use the other version

Reply:
bc it is obsolete